caffe2/core/operator.h

```diff
@@ -122,9 +122,24 @@ class CAFFE2_API OperatorBase : public Observable<OperatorBase> {
     static_assert(
         std::is_same<T, Tensor>::value,
         "Output(int, DeviceType) is only available for Tensor");
+    // When you get a Tensor here it is not fully initialized
     return BlobGetMutableTensor(outputs_.at(idx), type);
   }
 
+  template <typename T>
+  inline T* Output(
+      int idx,
+      const vector<int64_t>& dims,
+      const at::TensorOptions& options) {
+    static_assert(
+        std::is_same<T, Tensor>::value,
+        "Output(int, DeviceType) is only available for Tensor");
```

Comment:
So... we technically don't need the template parameter here at all, since this is a fresh overload. It seems a little goofy to say "You can have it in any color you want, as long as it's Tensor." Maybe consistency with the other Output methods justifies this, but I'm not sure.

Reply:
hmm, to avoid ambiguous call, we might need to change this function to virtual? which means we'll need to remove inline, which could potentially affect perf, let me run some benchmarks.

Reply:
OK I'll change the function name in the OperatorBase.